client/src/includes/bulk-actions.js

```diff
@@ -181,21 +181,38 @@ function onClickSelectAllInListing(e) {
  */
 function onClickActionButton(e) {
   e.preventDefault();
-  const url = e.target.getAttribute('href');
-  const urlParams = new URLSearchParams(window.location.search);
+  const currentURL = new URL(window.location.href);
+  const actionURL = new URL(e.target.getAttribute('href'), currentURL);
+
+  // Construct URL parameters for the action view
+  let urlParams;
   if (checkedState.selectAllInListing) {
+    // Preserve query parameters so the server can reconstruct the results
+    // when performing the action on all items in the current listing
+    urlParams = new URLSearchParams(currentURL.searchParams);
     urlParams.append('id', 'all');
     const parentElement = document.querySelector(BULK_ACTIONS_CHECKBOX_PARENT);
     if (parentElement) {
       const parentPageId = parentElement.dataset.bulkActionParentId;
       urlParams.append('childOf', parentPageId);
     }
   } else {
+    // No need to preserve existing query parameters as we're sending specific ids
+    urlParams = new URLSearchParams();
     checkedState.checkedObjects.forEach((objectId) => {
       urlParams.append('id', objectId);
     });
   }
-  window.location.href = `${url}&${urlParams.toString()}`;
+
+  // Use existing 'next' parameter if exists, otherwise use the current URL with
+  // query parameters to preserve any filters/search applied when redirecting back
+  const next =
+    currentURL.searchParams.get('next') ||
+    currentURL.pathname + currentURL.search;
+  urlParams.append('next', next);
```

Comment:
Unfortunately there's no JS tests for the bulk actions code, so you'll have to do manual testing to ensure this works.

Try:

1. Do a full page load of a listing
2. Apply search/filters/pagination
3. Select some items and click on a bulk action button
4. Upon confirming or cancelling the action, you should be taken back to the listing with the same search/filters/pagination applied

Might also want to try the following different scenarios:

- Have an existing `next` parameter, or
- Click "select all items in listing" instead of specific items, or
- Do a full page reload after applying the search/filters before clicking the bulk action